MAINTAINERS.md

```diff
@@ -89,31 +89,40 @@ A simple majority vote is required to authorize a SPIRE release at a specific co
 
 The following steps must be completed one week prior to release:
-* Ensure all changes intended to be included in the release are fully merged
-* Identify a specific commit as the release candidate
-* Raise an issue "Release SPIRE X.Y.Z", and include both the release candidate commit hash and the proposed changelog updates
+* Ensure all changes intended to be included in the release are fully merged.
+* Identify a specific commit as the release candidate.
+* Create a draft pull request with the updates to the CHANGELOG following [these guidelines](doc/changelog_guidelines.md). This allows to have publicly available the changes that will be included in the upcoming release with some anticipation, providing the oportunity to receive feedback. The release date can be set as "TBD" while it is a draft.
+* Raise an issue "Release SPIRE X.Y.Z", and include the release candidate commit hash. Reference the pull request with the updates to the CHANGELOG.
 
 **If this is a major release**, the following steps must be completed before releasing:
-* Review and exercise all examples in spiffe.io and spire-examples repo against the release candidate hash
-* Raise a PR for every example that updates included text and configuration to reflect current state and best practice
-  * Do not merge this PR yet. It will be updated later to use the real version pin rather than the commit hash
-  * If anything unusual is encountered during this process, a comment MUST be left on the release issue describing what was observed
+* Review and exercise all examples in spiffe.io and spire-examples repo against the release candidate hash.
+* Raise a PR for every example that updates included text and configuration to reflect current state and best practice.
+  * Do not merge this PR yet. It will be updated later to use the real version pin rather than the commit hash.
+  * If anything unusual is encountered during this process, a comment MUST be left on the release issue describing what was observed.
 
 The following steps must be completed to perform a release:
-* Cut two annotated tags against the release candidate named `vX.X.X` and `proto/spire/vX.X.X`, where `X.X.X` is the semantic version number of SPIRE
-  * The first line of the annotation should be `X.X.X` followed by the changelog. Refer to previous annotated tags as an example
+* Mark the pull request to update the CHANGELOG as "Ready for review". Make sure that it is updated with the final release date. **At least two approvals from maintainers are required in order to be able to merge it.**
+* If the current state of the master branch has diverged from the candidate commit due to just the CHANGELOG changes, the candidate commit is now the one that includes the updated CHANGELOG and the release can be done from master branch.
+* If the current state of the master branch has diverged from the candidate commit due to other changes than the ones from the CHANGELOG:
```

Comment:
Do we need to step in here anywhere to backport the CHANGELOG into master?

Reply:
My thinking was to have the CHANGELOG PR against master branch, merged in master and cherry-picked into the version branch. In the step:
>Cherry-pick into the version branch the commits for all the changes that must be included in the release, including the updates in the CHANGELOG and update the corresponding status in the project.

I tried to remind cherry-picking the updated CHANGELOG in the version branch saying "including the updates in the CHANGELOG".

Makes sense?